api_tests/nodes/views/test_node_list.py

```diff
@@ -96,6 +97,35 @@ def test_omit_retracted_registration(self):
         res = self.app.get(self.url, auth=self.user.auth)
         assert_equal(len(res.json['data']), 2)
 
+    def test_node_list_has_root(self):
+        res = self.app.get(self.url, auth=self.user.auth)
+        projects_with_root = 0
+        for project in res.json['data']:
+            if project['relationships'].get('root', None):
+                projects_with_root += 1
+        assert_equal(projects_with_root, 2)
+
+    def test_node_list_has_proper_root(self):
+        project_one = ProjectFactory(title="Project One", is_public=True)
+        ProjectFactory(parent=project_one, is_public=True)
+
+        res = self.app.get(self.url+'?embed=root&embed=parent', auth=self.user.auth)
+
+        projects_with_correct_root = 0
+        for project in res.json['data']:
+            root_id = project['embeds']['root']['data']['id']
+
+            if project['embeds']['parent'].get('data'):
+                parent_id = project['embeds']['parent']['data']['id']
+                if parent_id == project_one._id:
+                    if root_id == project_one._id:
```

Comment:
Rather than having just one assert at the end of this test, why not make this comparison an assertion:

```python
assert_equal(root_id, project_one._id)
```

That way, the test will fail early, and the output will tell you *which* root ID is incorrect.